lib/directory-tree.js

```diff
@@ -27,16 +27,18 @@ function directoryTree (path, options, onEachFile) {
 
 	try { stats = FS.statSync(path); }
 	catch (e) { return null; }
-	if (stats.isFile()) {
-		const ext = PATH.extname(path).toLowerCase();
 
-		// Only add files with the provided extensions
-		if (options && 
-			options.extensions && 
-			options.extensions.length && 
-			options.extensions.indexOf(ext) === -1)
-			return null;
+	// Skip if it matches the exclude regex
+	if (options && options.exclude && options.exclude.test(path))
+		return null;  
 
+	if (stats.isFile()) {
+		// Skip if it does not match the include regex
+		if (options && options.includeOnly && !options.includeOnly.test(path))
```

Comment:
@manatarms what if you extracted a helper function that had an extra flag argument signalling that it's traversing the root? You could call it with the flag `true` at first and make the following recursive calls with `false` and based on that you could look at `includeOnly` or not? What do you think?

Otherwise, sure, let's go with the extensions approach (maybe name the param accordingly) and see in the future if this becomes an ask.